main.go

```diff
@@ -66,22 +63,12 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
-	// NOTE: Pass in nil client if connection isn't established
-	var client controllers.PorterClient
-	conn, err := grpc.DialContext(context.Background(), "porter-grpc-service:3001", grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		setupLog.Error(err, "unable to set up listener")
-	}
-	defer conn.Close()
-	if conn != nil {
-		client = porterv1alpha1.NewPorterClient(conn)
-	}
 	if err = (&controllers.InstallationReconciler{
 		Client:           mgr.GetClient(),
-		PorterGRPCClient: client,
 		Recorder:         mgr.GetEventRecorderFor("installation"),
 		Log:              ctrl.Log.WithName("controllers").WithName("Installation"),
 		Scheme:           mgr.GetScheme(),
+		CreateGRPCClient: controllers.CreatePorterGRPCClient,
```

Comment:
This sets this to the controller but can be overridden in tests.  Scoped to the controller to allow for parallel testing if necessary.